std/stdio.d

```diff
@@ -4113,10 +4137,19 @@ private size_t readlnImpl(FILE* fps, ref char[] buf, dchar terminator = '\n')
         }
     }
 
-    char *lineptr = null;
-    size_t n = 0;
+    static char *lineptr = null;
+    static size_t n = 0;
+    scope(exit)
+    {
+        if (n > 128 * 1024)
+        {
+            // Bound memory used by readln
+            free(lineptr);
```

Comment:
Fortunately we have `std.stdiobase` for that, where we can include a call to the module destructor as well.